samples/cpp/text_generation/README.md

```diff
@@ -13,32 +24,99 @@ pip install --upgrade-strategy eager -r ../../requirements.txt
 optimum-cli export openvino --trust-remote-code --model TinyLlama/TinyLlama-1.1B-Chat-v1.0 TinyLlama-1.1B-Chat-v1.0
```

Comment:
I suppose we cannot use just one model for all use cases.

For chat sample - chat model
For typical generation - either instruct or typical model
For speculative decoding we need 2 models and it's not shown at all what models to use